src/docs/sphinx/theory_reference/odes.rst

```diff
@@ -0,0 +1,171 @@
+
+Now, let us consider how the solution process changes if some components
+of :math:`\bf{x}(t)` are controlled externally. Now our differential
+equation looks like
+
+.. math::
+
+   \bf{M} \; \ddot{\bf{x}}(t) + \bf{C} \; \dot{\bf{x}}(t) + \bf{K} \; \bf{x}(t)  = \bf{0} \\
+
+   \color{red}{\bf{G} \; \bf{x}(t) = \bf{g}(t)} \\
+
+   \bf{x}(0) = \bf{x}_0 \\
+
+   \dot{\bf{x}}(0) = \dot{\bf{x}}_0
+
+where :math:`\bf{G}` is a matrix with 1 on the diagonals corresponding
+to constrained components, and 0 elsewhere, and :math:`\bf{g}(t)` is
+the vector of prescribed values for the constrained degrees of freedom.
+In this case, there are several equivalent ways to interpret the
+additional constraint equations (we assume the constraints are
+consistent with the initial conditions):
+
+.. math::
+
+   \bf{G} \; \bf{x}(t) = \bf{g}(t) \; \iff \;
+
+   \bf{G} \; \dot{\bf{x}}(t) = \dot{\bf{g}}(t) \; \iff \;
+
+   \bf{G} \; \ddot{\bf{x}}(t) = \ddot{\bf{g}}(t)
+
+At the end of the day, we are solving for a single acceleration vector,
+:math:`\ddot{\bf{x}}(t+\Delta t)`, and as a result, we cannot hope to
+simultaneously satisfy more than one interpretation of these
+constraints. The different Dirichlet Enforcement Methods in serac relate
+to the following different interpretations of the constraint equations.
+
+.. _header-n97:
+
+"Direct Control"
+----------------
+
+Direct control proceeds to solve for accelerations by effectively
+replacing the constrained rows the differential equation by the
+constraint equations,
+
+.. math:: (\bf{1} - \bf{G})(\bf{M} \; \ddot{\bf{x}}(t+\Delta t) + \bf{C} \; \dot{\bf{x}}(t+\Delta t) + \bf{K} \; \bf{x}(t+\Delta t)) + \bf{G} \; \bf{x}(t+\Delta t)\;=\;\bf{g}(t+\Delta t)
+
+Now, like before, we substitute in expressions for extrapolated state
+and solve for :math:`\ddot{\bf{x}}(t+\Delta t)`. In this
+interpretation of the constraint, the resulting accelerations ensure
+that :math:`\bf{G} \; \bf{x}(t+\Delta t) = \bf{g}(t+\Delta t)` is
+satisfied.
+
+.. _header-n79:
+
+"Rate Control"
+--------------
+
+Rate control uses a similar idea, but with the rate-form interpretation
+of the constraint:
+
+.. math:: (\bf{1} - \bf{G})(\bf{M} \; \ddot{\bf{x}}(t+\Delta t) + \bf{C} \; \dot{\bf{x}}(t+\Delta t) + \bf{K} \; \bf{x}(t+\Delta t)) + \bf{G} \; \dot{\bf{x}}(t+\Delta t)\;=\;\dot{\bf{g}}(t+\Delta t)
+
+Here, the end-step accelerations are determined such that the derivative
+of the constraint equations is exactly satisfied,
+:math:`\bf{G} \; \dot{\bf{x}}(t+\Delta t) = \dot{\bf{g}}(t+\Delta t)`.
+This tends to have better stability than the "Direct Control" option,
+but is prone to solution "drift" over long periods of time.
+
+.. _header-n94:
+
+"Full Control"
+--------------
+
+This last option starts by additively decomposing the solution vector
+into two parts: constrained and unconstrained:
+
+.. math::
+
+   \begin{cases}
+
+   \bf{x}(t) = \bf{x}_c(t) + \bf{x}_u(t) = \bf{G} \; \bf{x}(t) + (\bf{1} - \bf{G}) \; \bf{x}(t) \\
```

Comment:
Minor: It might be clearer if the distinction between constrained/unconstrained is preserved in the third interpretation of x(t) and its derivatives.  To me, the actual implementation looks like `G * x_c(t) + (1 - G) * x_u(t)` aka `G * g(t) + (1 - G) * x_u(t)`.